src/controllers/controllerengine.cpp

```diff
@@ -185,18 +185,18 @@ void ControllerEngine::gracefulShutdown() {
         }
     }
 
-    // Clear the cache of function wrappers
+    qDebug() << "Clearing function wrapper cache";
     m_scriptWrappedFunctionCache.clear();
 
     // Free all the ControlObjectScripts
-    QList<ConfigKey> keys = m_controlCache.keys();
-    QList<ConfigKey>::iterator it = keys.begin();
-    QList<ConfigKey>::iterator end = keys.end();
-    while (it != end) {
-        ConfigKey key = *it;
-        ControlObjectScript* coScript = m_controlCache.take(key);
-        delete coScript;
-        ++it;
+    for (auto it = m_controlCache.begin(); it != m_controlCache.end(); /*see loop body*/) {
```

Comment:
Cpp has while loops fore situation where the iteration is done in the loop body.